### PR TITLE
Improve the error message when the profiler can't be initialized

### LINKED ISSUE #2379: Application insight profiler is not working in Spring boot App
### Expected behavior
Profiler should run

### Actual behavior
Getting following exception 
`2022-07-06 09:21:52.725Z ERROR c.m.a.a.i.profiler.GcEventMonitor - Failed to monitor gc mxbeans
2022-07-06 09:21:52.749Z WARN  c.m.a.s.JfrProfilerService - INITIALISING JFR PROFILING SUBSYSTEM THIS FEATURE IS IN BETA
2022-07-06 09:21:52.832Z ERROR c.m.a.s.profiler.JfrProfiler - Failed to connect to mbean
javax.management.InstanceNotFoundException: com.sun.management:type=DiagnosticCommand
	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getMBean(DefaultMBeanServerInterceptor.java:1083)
	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getObjectInstance(DefaultMBeanServerInterceptor.java:458)
	at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.getObjectInstance(JmxMBeanServer.java:564)
	at com.microsoft.jfr.dcmd.FlightRecorderDiagnosticCommandConnection.connect(FlightRecorderDiagnosticCommandConnection.java:76)
	at com.microsoft.applicationinsights.serviceprofilerapi.profiler.JfrProfiler.initialize(JfrProfiler.java:164)
	at com.microsoft.applicationinsights.serviceprofilerapi.JfrProfilerService.initialiseProfiler(JfrProfilerService.java:143)
	at com.microsoft.applicationinsights.serviceprofilerapi.JfrProfilerService.lambda$initialize$0(JfrProfilerService.java:114)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:871)
2022-07-06 09:21:52.833Z ERROR c.m.a.s.JfrProfilerService - Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
2022-07-06 09:21:52.835Z ERROR c.m.a.a.i.p.ProfilerServiceInitializer - Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
java.util.concurrent.ExecutionException: java.lang.RuntimeException: Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
	at com.microsoft.applicationinsights.agent.internal.profiler.ProfilerServiceInitializer.lambda$initialize$0(ProfilerServiceInitializer.java:151)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:871)
Caused by: java.lang.RuntimeException: Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
	at com.microsoft.applicationinsights.serviceprofilerapi.JfrProfilerService.lambda$initialize$0(JfrProfilerService.java:115)
	... 6 common frames omitted`

### To Reproduce
Enable application insight profiler in spring boot app using applicationinsight.json config using following code
` "preview": {
    "profiler":{
      "configPollPeriodSeconds": 60,
      "enabled":true
    },`

### System information
Please provide the following information:
 - SDK Version: 3.3.0
 - OS type and version: linux
 - Application Server type and version (if applicable):
 - Using spring-boot? yes
 - Additional relevant libraries (with version, if applicable):

### Logs
`2022-07-06 09:21:52.725Z ERROR c.m.a.a.i.profiler.GcEventMonitor - Failed to monitor gc mxbeans
2022-07-06 09:21:52.749Z WARN  c.m.a.s.JfrProfilerService - INITIALISING JFR PROFILING SUBSYSTEM THIS FEATURE IS IN BETA
2022-07-06 09:21:52.832Z ERROR c.m.a.s.profiler.JfrProfiler - Failed to connect to mbean
javax.management.InstanceNotFoundException: com.sun.management:type=DiagnosticCommand
	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getMBean(DefaultMBeanServerInterceptor.java:1083)
	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getObjectInstance(DefaultMBeanServerInterceptor.java:458)
	at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.getObjectInstance(JmxMBeanServer.java:564)
	at com.microsoft.jfr.dcmd.FlightRecorderDiagnosticCommandConnection.connect(FlightRecorderDiagnosticCommandConnection.java:76)
	at com.microsoft.applicationinsights.serviceprofilerapi.profiler.JfrProfiler.initialize(JfrProfiler.java:164)
	at com.microsoft.applicationinsights.serviceprofilerapi.JfrProfilerService.initialiseProfiler(JfrProfilerService.java:143)
	at com.microsoft.applicationinsights.serviceprofilerapi.JfrProfilerService.lambda$initialize$0(JfrProfilerService.java:114)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:871)
2022-07-06 09:21:52.833Z ERROR c.m.a.s.JfrProfilerService - Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
2022-07-06 09:21:52.835Z ERROR c.m.a.a.i.p.ProfilerServiceInitializer - Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
java.util.concurrent.ExecutionException: java.lang.RuntimeException: Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
	at com.microsoft.applicationinsights.agent.internal.profiler.ProfilerServiceInitializer.lambda$initialize$0(ProfilerServiceInitializer.java:151)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:871)
Caused by: java.lang.RuntimeException: Unable to obtain JFR connection, this may indicate that your JVM does not have JFR enabled. JFR profiling system will shutdown
	at com.microsoft.applicationinsights.serviceprofilerapi.JfrProfilerService.lambda$initialize$0(JfrProfilerService.java:115)
	... 6 common frames omitted`

_Be sure to remove any private information from the logs before posting!_

### PR DESCRIPTION
**Pull request by jeanbisutti**
Improve the error message when the profiler can't be initialized

Fix #2379

### CODE REVIEW

File: agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java [hunk 1]
```diff
@@ -133,13 +133,23 @@ public boolean initialize(
...
+      LOGGER.error(message, e);
       return false;
     }
 
     return true;
   }
 
+  private static boolean isOpenJ9Jvm() {
+    String jvmName = System.getProperty("java.vm.name");
+    return jvmName.contains("OpenJ9");
```
Comment (heyams) [inline agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java:150]:
```suggestion
    return jvm != null && jvmName.contains("OpenJ9");
```

File: agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java [hunk 2]
```diff
@@ -133,13 +133,23 @@ public boolean initialize(
         flightRecorderConnection = FlightRecorderDiagnosticCommandConnection.connect(mbeanServer);
       }
     } catch (Exception e) {
-      LOGGER.error("Failed to connect to mbean", e);
+      String message =
+          "Unable to initialize the profiler."
+              + (isOpenJ9Jvm()
+                  ? " OpenJ9 JVM is not supported. Instead, please use an OpenJDK JVM."
+                  : "");
+      LOGGER.error(message, e);
```
Comment (trask) [inline agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java:141]: I'd suggest instead checking for OpenJ9 proactively before initializing the profiler

File: agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java (hunk 1 above)
Comment (jeanbisutti) [reply agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java]: Integrated, not sure jvmName could be null

File: agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java (hunk 2 above)
Comment (jeanbisutti) [reply agent/agent-profiler/agent-service-profiler/src/main/java/com/microsoft/applicationinsights/serviceprofilerapi/profiler/JfrProfiler.java]: Done; the drawback of doing this is that it won't become possible to benefit from the current initialization code if JFR is one day integrated into OpenJ9.

**Review by johnoliver — APPROVED** [review]

**Pull request closed without merging**